.github/workflows/test-demo.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+DISTRO="$1"
+NIXPKGS="https://github.com/NixOS/nixpkgs/archive/nixos-unstable.tar.gz"
+# shellcheck disable=SC2089,2026
+NIX_CONFIG='substituters = https://cache.nixos.org/ https://ngi.cachix.org/'$'\n''trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= ngi.cachix.org-1:n+CAL72ROC3qQuLxIHpV+Tw5t42WhXmMhprAGkRSrOw='
+
+export NIX_CONFIG
+
+echo -e "\n-> Installing Nix ..."
+# Debian/Ubuntu
+if echo "$DISTRO" | grep --quiet "debian\|ubuntu"; then
+    apt update
+    apt install --yes curl git nix
+# Archlinux
+elif echo "$DISTRO" | grep --quiet archlinux; then
+    pacman --sync --refresh --noconfirm curl git nix
+# Other
```

Comment:
Maybe add alpine and void, as per [this issue](https://github.com/ngi-nix/ngipkgs/issues/697).